code/modules/projectiles/guns/energy/hos.dm

```diff
@@ -0,0 +1,304 @@
+// -------------- HoS Modular Weapon System -------------
+// ---------- Code originally from VoreStation ----------
+/obj/item/gun/ballistic/revolver/mws
+	name = "MWS-01 'Big Iron'"
+	desc = "Modular Weapons System"
+
+	icon = 'icons/obj/guns/projectile.dmi'
+	icon_state = "mws"
+
+	fire_sound = 'sound/weapons/Taser.ogg'
+
+	mag_type = /obj/item/ammo_box/magazine/mws_mag
+	spawnwithmagazine = FALSE
+
+	recoil = 0
+
+	var/charge_sections = 6
+
+/obj/item/gun/ballistic/revolver/mws/afterattack(atom/target, mob/living/user, flag, params)
+	if(chambered && !chambered.BB) //if BB is null, i.e the shot has been fired...
+		var/obj/item/ammo_casing/mws_batt/shot = chambered
+		if(shot.shots_left >= 1)
+			shot.newshot() //Create a new shot
+			update_charge()
+	.=..()
+
+/obj/item/gun/ballistic/revolver/mws/process_chamber()
+	if(chambered && !chambered.BB) //if BB is null, i.e the shot has been fired...
+		var/obj/item/ammo_casing/mws_batt/shot = chambered
+		if(shot.shots_left >= 1)
+			shot.newshot() //Create a new shot
+			update_charge()
+		else
+			for(var/B in magazine.stored_ammo)
+				var/obj/item/ammo_casing/mws_batt/other_batt = B
+				if(istype(other_batt,shot) && other_batt.shots_left >= 1)
+					switch_to(other_batt)
+	else
+		return
+
+/obj/item/gun/ballistic/revolver/mws/proc/update_charge()
+
+	if(!chambered)
+		return
+
+	var/obj/item/ammo_casing/mws_batt/batt = chambered
+	if(batt.cell.charge >= batt.e_cost)
```

Comment:
replace all of this and the whole .newshot() part with a proc on the battery to use the energy cost of a shot
also why does this do this on update_charge? update charge implies updating not using charge to use a new shot